### ISSUE TITLE
Working on RelaxNG tooling

### ISSUE THREAD

**Issue by skiadas** [skiadas/rngast#1]
@oscarlevin  I've been working on unifiedjs tooling for RelaxNG grammars, I forget when/where we were discussing this (drop-in maybe?). Still much more to be done but I had enough that I thought it was time to start sharing it. Hence this repo.

**Comment by oscarlevin** [comment]
Very cool!  You might also want to take a peek at https://github.com/siefkenj/relax-ng-to-typescript.  I haven't looked at that our what you have here closely enough to know if there is overlap or whether they can support each other.  This is all very exciting.

**Comment by skiadas** [comment]
> Very cool! You might also want to take a peek at https://github.com/siefkenj/relax-ng-to-typescript. I haven't looked at that our what you have here closely enough to know if there is overlap or whether they can support each other. This is all very exciting.

Ah interesting, yes I see a lot of overlap on a quick glance! I'm guessing this is related to @siefkenj work on pretext on the browser! I need to check that out more.

**Comment by siefkenj** [comment]
Yep :-). This is how I get schema information exported. I implemented most of the steps of the relax-ng normalization algorithm. Maybe you did the same?

I didn't get so far as to parse the simplified rng syntax, so I rely on the XML form from the start.

**Comment by skiadas** [comment]
> Yep :-). This is how I get schema information exported. I implemented most of the steps of the relax-ng normalization algorithm. Maybe you did the same?

Yeah, it's not quite complete yet, I am missing some of the "simple" steps, things to do with namespaces, and a lot of testing. But I didn't think of generating types at the end of the day, I was looking at just writing a parser/validator for a ptx-document-as-xast given a relax-ng xml grammar file. I would like to see if it can be used to provide context-specific autocomplete features for Oscar's VSCode extension.
 
> I didn't get so far as to parse the simplified rng syntax, so I rely on the XML form from the start.

Same. Though it sure would be nice to read and write the simplified syntax.
